### PR TITLE
Fix R bindings for  MacOS

### PR DESCRIPTION
**Pull request by lmoneta**
This PR fixes a problem found when building and using the R bindings on MacOS

- First we make sure that ROOT is not using the old RConfig.h but ROOT/RConfig.hxx
  This avoids a conflict with the R file Rconfig.h
  The first commit remove the usage of the old Rconfig.h when generating the dictionary for libCore 

- The second commit fixes a conflicting definition of a macro between system macOS and R headers.

### CODE REVIEW

File: bindings/r/inc/RExports.h [hunk 1]
```diff
@@ -48,7 +64,7 @@
 //Some useful typedefs
 typedef std::vector<TString> TVectorString;
 
-#include<RcppCommon.h>
+#
```
Comment (Axel-Naumann) [inline bindings/r/inc/RExports.h:67]: This seems wrong

**Review by Axel-Naumann — CHANGES_REQUESTED** [review]
Leftover `#`

File: bindings/r/inc/RExports.h (hunk 1 above)
Comment (lmoneta) [reply bindings/r/inc/RExports.h]: Thanks for spotting this. It is fixed now

**Comment by lmoneta** [comment]
@Axel-Naumann : I have added a new commit fixing the compilation warnings seen when compiling the R interface and also when including at the ROOT prompt a file like TRInterface.h

**Comment by lmoneta** [comment]
@phsft-bot build just on ROOT-performance-centos8-multicore/default with flags -Dr=ON

**Comment by lmoneta** [comment]
phsft-bot build with flags -Dr=On

File: bindings/r/inc/RExports.h [hunk 2]
```diff
@@ -43,12 +46,27 @@
 #pragma GCC diagnostic ignored "-Wextra"
 #pragma GCC diagnostic ignored "-Wignored-qualifiers"
 #endif
+// disable warning for macos
+#if defined(__APPLE__)
+#pragma GCC diagnostic ignored "-Wnonportable-include-path"
+#
```
Comment (Axel-Naumann) [inline bindings/r/inc/RExports.h:52]: That needs to be "undone" below, after the relevant `#include`, else nobody including this header will get this warning anymore.

File: build/unix/module.modulemap [hunk 3]
```diff
@@ -14,8 +14,6 @@ module ROOT_Foundation_C  {
 module ROOT_Config  {
   // These headers are supposed to be only textually expanded for each TU.
   module "RVersion.h" { textual header "RVersion.h" export * }
-  module "RConfig.h" { header "RConfig.h" export * }
-  module "ROOT/RConfig.h" { header "ROOT/RConfig.h" export * }
```
Comment (Axel-Naumann) [inline build/unix/module.modulemap:18]: @vgvassilev these clash with headers from R with the same name. Do we really need these (they just fwd include to `ROOT/RConfig.hxx`) in the module?

**Review by Axel-Naumann — CHANGES_REQUESTED** [review]

File: bindings/r/inc/RExports.h (hunk 2 above)
Comment (lmoneta) [reply bindings/r/inc/RExports.h:52]: I agree, but then should we do this also for all other warnings that are ignored ? See the lines above.

**Comment by lmoneta** [comment]
@phsft-bot build with flags -Dr=On -Dfail-on-missing=Off

File: build/unix/module.modulemap (hunk 3 above)
Comment (vgvassilev) [reply build/unix/module.modulemap:18]: Do we need `RConfig.h` you mean?

Comment (lmoneta) [reply build/unix/module.modulemap:18]: Why ? In principle no ROOT file will include RConfig.h but they will use ROOT/RConfig.hxx

File: bindings/r/inc/RExports.h (hunk 2 above)
Comment (Axel-Naumann) [reply bindings/r/inc/RExports.h]: Great thanks!

File: build/unix/module.modulemap (hunk 3 above)
Comment (lmoneta) [reply build/unix/module.modulemap]: I don't think many people outside ROOT include this file. They include ROOT headers that include ROOT/RConfig.hxx. And probably the duplication is just for a forward declaration, so it should not be a big thing.
Anyway, I have reverted those changes, probably this was not the cause of the conflict seen before, It was caused by the inclusion of  Rconfig.h in G__Core.cxx, that is now fixed. 
Let's see what will be  the result of the CI builds

**Review by Axel-Naumann — APPROVED** [review]

**Pull request merged**